### PR TITLE
Cultist pets

### PR DESCRIPTION
**Pull request by Ben10Omintrix**
## About The Pull Request
cultists can now convert pets on their side. when you convert a pet, it will become much more dangerous and obedient to you, it will also gain a new dangerous AI. cultist pets will look for fellow dead cultists, and revive them by dragging them to nearby revival runes and activating it. if there is no revival runes around them then they will create their own and drag u to it. u can give them commands to follow, attack, or to create revival runes. they will also go around to crit non cultists so cultists can convert them
pughealed

they will also now feed on organs and blood for healing

## Why It's Good For The Game
adds a extra layer to cult, u can now command non-sentient pets to aid you

## Changelog
:cl:
add: cultists can now convert pets to their side
/:cl:

### CODE REVIEW

**Comment by Loafin34** [comment]
yet another banger created by the indomitable ben10omnitrix, will you be able to convert cargorilla and lia/sergeant araneus?

**Comment by Draggeru** [comment]
This is adorable and I love it, if you want sprites for this hit me up on discord and Ill throw Cre at it.

(Cre is the spriter for the Tape Wizard outfit)

**Review by MrMelbert — CHANGES_REQUESTED** [review]
Do not let them help invoke runes if they are not player controlled, learn from the mistakes of the past when you could solo convert with a Nar'sie plushie

File: code/modules/antagonists/cult/runes.dm [hunk 1]
```diff
@@ -222,38 +224,37 @@ structure_check() searches for nearby cultist structures required for the invoca
...
-	var/datum/antagonist/cult/C = F.mind.has_antag_datum(/datum/antagonist/cult,TRUE)
-	var/datum/team/cult/Cult_team = C.cult_team
-	var/is_convertable = is_convertable_to_cult(L, C.cult_team)
-	if(L.stat != DEAD && is_convertable)
-		invocation = "Mah'weyh pleggh at e'ntrath!"
-		..()
-		if(is_convertable)
-			do_convert(L, invokers, Cult_team)
+	var/mob/living/selected_victim = pick(myriad_targets)
+	var/datum/team/cult/cult_team =  locate(/datum/team/cult) in GLOB.antagonist_teams
```
Comment (MrMelbert) [inline code/modules/antagonists/cult/runes.dm:244]: Locating in global teams is weird, the old code was fine (https://github.com/tgstation/tgstation/pull/80583/files#diff-03b369e5e6c74160bc7eb992d2b7fc282e6f07d67d6669ef6d5e7f25458dcc9f)
```dm
var/mob/living/new_convertee = pick(myriad_targets)
var/mob/living/first_invoker = invokers[1]
var/datum/antagonist/cult/first_invoker_datum = first_invoker.mind.has_antag_datum(/datum/antagonist/cult)
var/datum/team/cult/cult_team = first_invoker_datum.get_team()
```

Comment (Ben10Omintrix) [reply code/modules/antagonists/cult/runes.dm:244]: I initially needed to do this incase my invoker didn’t have a mind

**Comment by Jacquerel** [comment]
> yet another banger created by the indomitable ben10omnitrix, will you be able to convert cargorilla and lia/sergeant araneus?

These animals do not need converting and are already basic mobs, although anyone can always mess with the ai on any basic mob whenever they feel like it.
Cargorilla though notably never appears with AI in the first place, it's exclusively player-controlled.

**Comment by itseasytosee** [comment]
Can we not already solo convert really easy with the summoning rune?

**Comment by Ben10Omintrix** [comment]
They will automatically listen to the commands of any nearby cultists

**Comment by Ben10Omintrix** [comment]
now instead of converting people, they will revive dead cultists by using revival runes

**Comment by Draggeru** [comment]
> now instead of converting people, they will revive dead cultists by using revival runes

Are you making a special revival rune type thatll always work? Because I see the pets getting stuck in a logic loop trying to trigger the rune over and over while there are no ample sacrifices for the rune

**Comment by Ben10Omintrix** [comment]
They'll only try to revive people if there's enough sacrifices. otherwise they won't

**Review by Wallemations — APPROVED** [review]
finally, we can have cute murder cults

**Comment by ExcessiveUseOfCobblestone** [comment]
> Can we not already solo convert really easy with the cult ghost rune?

ghost rune comes with tradeoffs especially if you get at least 2 (intended)

**Comment by Admiral-Gigaton** [comment]
Feature freeze january?

also, does this work on cargorilla? if so, what happens to player control and the pacifism?

does this only work on animals that are tamed? or only on round-start pets? could i mass-convert carp? and also can i convert pet carp (HoS' carp, lia)

i think we should be able to convert with these, i mean you can alreayd ghost rune it, and theres only so many players with access to getting/taming pets to convert.

also, NARS'IAN CAN FINALLY BE OBTAINED NORMALLY NOW? WOOOO

**Comment by ExcessiveUseOfCobblestone** [comment]
is this waiting on review or?

**Comment by Ben10Omintrix** [comment]
ive now returned to working on this

**Comment by Ben10Omintrix** [comment]
this is now ready

**Comment by Ghommie** [comment]
Make sure it compiles. I'll review it soon.

**Comment by Ben10Omintrix** [comment]
its now ready! (again)

**Comment by optimumtact** [comment]
>I'll review it soon.

File: code/modules/antagonists/cult/runes.dm [hunk 2]
```diff
@@ -689,50 +690,50 @@ GLOBAL_VAR_INIT(narsie_summon_count, 0)
...
 	if(QDELETED(src) || !validness_checks(mob_to_revive, user))
 		fail_invoke()
 		return
-	if(user.name == "Herbert West")
-		invocation = "To life, to life, I bring them!"
-	else
-		invocation = initial(invocation)
-	..()
+
+	invocation = (user.name == "Herbet West") ? "To life, to life, I bring them!" : initial(invocation)
```
Comment (san7890) [inline code/modules/antagonists/cult/runes.dm:728]:
```suggestion
	invocation = (user.name == "Herbert West") ? "To life, to life, I bring them!" : initial(invocation)
```

don't ruin the joke with a typo in the refactor!

File: code/modules/mob/living/basic/pets/pet_cult/pet_cult_abilities.dm [hunk 3]
```diff
@@ -0,0 +1,14 @@
+/datum/action/cooldown/spell/conjure/convert_rune
+	name = "Create Conversion Rune"
+	button_icon = 'icons/obj/antags/cult/rune.dmi'
+	button_icon_state = "1"
+	background_icon_state = "bg_cult"
+	overlay_icon_state = "bg_cult_border"
+	spell_requirements = NONE
+	cooldown_time = 30 SECONDS
+	summon_type = list(
+		/obj/effect/rune/raise_dead
```
Comment (san7890) [inline code/modules/mob/living/basic/pets/pet_cult/pet_cult_abilities.dm:10]:
```suggestion
		/obj/effect/rune/raise_dead,
```

File: code/modules/mob/living/basic/pets/pet_cult/pet_cult_ai.dm [hunk 4]
```diff
@@ -0,0 +1,246 @@
+/datum/ai_controller/basic_controller/pet_cult
+	blackboard = list(
+		BB_TARGETING_STRATEGY = /datum/targeting_strategy/basic/cultist,
+		BB_PET_TARGETING_STRATEGY = /datum/targeting_strategy/basic/cultist,
+		BB_FRIENDLY_MESSAGE = "eagerly awaits your command..."
```
Comment (san7890) [inline code/modules/mob/living/basic/pets/pet_cult/pet_cult_ai.dm:5]:
```suggestion
		BB_FRIENDLY_MESSAGE = "eagerly awaits your command...",
```

**Review by san7890 — CHANGES_REQUESTED** [review]
i just scanned through some of it. few nits but one big big code consideration

File: code/modules/antagonists/cult/runes.dm [hunk 5]
```diff
@@ -161,17 +160,19 @@ structure_check() searches for nearby cultist structures required for the invoca
...
-				to_chat(L, "<span class='cult italic'>[src] saps your strength!</span>")
-		else if(istype(M, /obj/item/toy/plush/narplush))
-			var/obj/item/toy/plush/narplush/P = M
-			P.visible_message("<span class='cult italic'>[P] squeaks loudly!</span>")
+	for(var/atom/invoker in invokers)
+		if(istype(invoker, /obj/item/toy/plush/narplush))
+			invoker.visible_message(span_cult_italic("[src] squeaks_loudly!"))
+			continue
+		if(!isliving(invoker))
+			return
```
[the remark this answers is not in the document]
Comment (Ben10Omintrix) [reply code/modules/antagonists/cult/runes.dm:168]: woop

File: code/modules/mob/living/basic/pets/pet_cult/pet_cult_ai.dm [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+/datum/ai_planning_subtree/find_occupied_rune/SelectBehaviors(datum/ai_controller/controller, seconds_per_tick)
+	if((LAZYLEN(GLOB.sacrificed) - SOULS_TO_REVIVE - GLOB.sacrifices_used) < 0)
+		controller.clear_blackboard_key(BB_OCCUPIED_RUNE)
+		return
+
+	if(controller.blackboard_key_exists(BB_OCCUPIED_RUNE))
+		controller.queue_behavior(/datum/ai_behavior/activate_rune, BB_OCCUPIED_RUNE)
+		return SUBTREE_RETURN_FINISH_PLANNING
+
+	controller.queue_behavior(/datum/ai_behavior/find_and_set/occupied_rune, BB_OCCUPIED_RUNE, /obj/effect/rune/convert)
```
Comment (Ghommie) [inline code/modules/mob/living/basic/pets/pet_cult/pet_cult_ai.dm:71]:
```suggestion
	controller.queue_behavior(/datum/ai_behavior/find_and_set/occupied_rune, BB_OCCUPIED_RUNE, /obj/effect/rune/raise_dead)
```

File: code/modules/mob/living/basic/pets/pet_cult/pet_cult_ai.dm [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+	controller.queue_behavior(/datum/ai_behavior/find_and_set/dead_cultist, BB_DEAD_CULTIST, /mob/living/carbon/human)
+
+/datum/ai_behavior/find_and_set/dead_cultist
+
+/datum/ai_behavior/find_and_set/dead_cultist/search_tactic(datum/ai_controller/controller, locate_path, search_range)
+	var/datum/team/cult/cult_team = controller.blackboard[BB_CULT_TEAM]
+	if(isnull(cult_team))
+		return null
+	var/mob/living/living_pawn = controller.pawn
+	for(var/mob/living/carbon/human/target in oview(search_range, controller.pawn))
```
Comment (Ghommie) [inline code/modules/mob/living/basic/pets/pet_cult/pet_cult_ai.dm:154]:
```suggestion
	for(var/mob/living//target in oview(search_range, controller.pawn))
```

The revive rune can be used on all living mobs that have the cultist antag datum.

**Review by Ghommie — CHANGES_REQUESTED** [review]

**Review by L0pz — APPROVED** [review]
Please please please make it so poly spouts cultist stuff over the radio

**Pull request merged**